numpy/core/numeric.py

```diff
@@ -58,6 +58,22 @@
     '_get_promotion_state', '_set_promotion_state']
 
 
+# TODO: Remove this wrapper when fastCopyAndTranspose deprecation expires
+def fastCopyAndTranspose(a):
+    """
+    .. deprecated:: 1.24
+
+       fastCopyAndTranspose is deprecated and will be removed. Use the copy
+       and transpose methods instead, e.g. ``arr.copy().T``
+    """
+    warnings.warn(
+        "fastCopyAndTranspose is deprecated. Use ``arr.copy().T`` instead",
```

Comment:
```suggestion
        "fastCopyAndTranspose is deprecated. Use ``arr.T.copy()`` instead",
```
You might also want to mention the "order" argument?

It desn't need much (not C code), but it would be nice to test that the warning is given.  One option: Just keep the old tests and add `with pytest.warns(DeprecationWarning):`, rather than adding a new test.

Reply:
> One option: Just keep the old tests and add with pytest.warns(DeprecationWarning):, rather than adding a new test.

The reason I did it this way was that the "private" `_fastCopyAndTranspose` function is still used, so I reorganized the tests so that the private function is tested instead. I will add back the tests for the public function w/ the warnings catcher, as you're right that those tests should only be removed when the deprecation expires!

Also :+1: for the suggestion - I need to add it to the deprecation directive in the docs too...

Reply:
Okay the deprecation message is updated in 7b43315 and the tests of the public fn (with the deprecation check) are added back in bd9236d.

I didn't know how best to mention the `order` kwarg without going into too much detail for a DeprecationWarning, so I simply went with `order='K'` - LMK what you think!